src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/CustomConverterTests/CustomConverterTests.Callback.cs

```diff
@@ -152,10 +151,40 @@ public static void ConverterWithReentryFail()
             };
 
             ex = Assert.Throws<NotSupportedException>(() => JsonSerializer.Serialize(poco, options));
-            Assert.Contains(typeof(int[,]).ToString(), ex.ToString());
-            Assert.Contains(typeof(ChildPocoWithNoConverterAndInvalidProperty).ToString(), ex.ToString());
-            Assert.Contains("Path: $.InvalidProperty.", ex.ToString());
+            Assert.Contains(typeof(int[,]).ToString(), ex.Message);
+            Assert.Contains("Path: $.InvalidProperty.NotSupported.", ex.Message);
             Assert.Equal(2, ex.ToString().Split(new string[] { "Path:" }, StringSplitOptions.None).Length);
         }
+
+        [Fact]
+        public static void UnsupportedPropertyWithCustomConverter()
+        {
+            // Regression test for https://github.com/dotnet/runtime/issues/80914
+            var dto = new PocoWithConverterOnInvalidProperty { NotSupported = new int[,] { { 1, 0 }, { 0, 1 } } };
+
+            string json = JsonSerializer.Serialize(dto);
+            Assert.Equal("""{"NotSupported":null}""", json);
+
+            dto = JsonSerializer.Deserialize<PocoWithConverterOnInvalidProperty>(json);
+            Assert.Null(dto.NotSupported);
+        }
+
+        private class PocoWithConverterOnInvalidProperty
+        {
+            [JsonConverter(typeof(MultiDimArrayConverter))]
+            public int[,] NotSupported { get; set; }
+
+            public class MultiDimArrayConverter : JsonConverter<int[,]>
+            {
+                public override int[,]? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+                {
+                    reader.Skip();
+                    return null;
+                }
+
+                public override void Write(Utf8JsonWriter writer, int[,] value, JsonSerializerOptions options)
+                    => writer.WriteNullValue();
```

Comment:
can we write something else than null here? Even int with specific value will be fine, null seems like something which might happen by accident

Reply:
Done.